pkg/cmd/cli/restic/server.go

```diff
@@ -191,3 +198,40 @@ func (s *resticServer) run() {
 	s.logger.Info("Waiting for all controllers to shut down gracefully")
 	wg.Wait()
 }
+
+// validatePodVolumesHostPath validates that the podVolumesPath contains a
+// directory for each Pod running on this node
+func validatePodVolumesHostPath(kubeClient kubernetes.Interface, podVolumesPath string) error {
+	files, err := ioutil.ReadDir(podVolumesPath)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	// create a map of directory names inside the podVolumesPath
+	dirs := map[string]bool{}
+	for _, f := range files {
+		if f.IsDir() {
+			dirs[f.Name()] = true
+		}
+	}
+
+	pods, err := kubeClient.CoreV1().Pods("").List(metav1.ListOptions{FieldSelector: fmt.Sprintf("spec.nodeName=%s", os.Getenv("NODE_NAME"))})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	for _, pod := range pods.Items {
+		dirName := string(pod.GetUID())
+
+		// if the pod is a mirror pod, the directory name is the hash value of the
+		// mirror pod annotation
+		if hash, ok := pod.GetAnnotations()[v1.MirrorPodAnnotationKey]; ok {
+			dirName = hash
+		}
+
+		if _, ok := dirs[dirName]; !ok {
+			return errors.WithStack(fmt.Errorf("could not find volumes for pod %s/%s in hostpath, check if the pod volumes hostPath mount is correct", pod.GetNamespace(), pod.GetName()))
```

Comment:
I'm wondering if instead of returning immediately the first time we don't find a pod, we should collect all the pods that aren't found, and then return an error listing all of them? WDYT?

Reply:
So actually I was thinking that it might not even make sense to display the name of the Pod volume that's missing, as most likely the issue is that the host mount is wrong. It might be less confusing to just have the error here be "unexpected directory structure for pod volumes mount, check if the hostPath mount is correct".

After discussing at standup, @skriss suggested debug logging each missing pod volume too in case a user is trying to debug why the host mount is wrong.